core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -5530,6 +5530,58 @@ private void checkSemiJoinRuleOnAntiJoin(RelOptRule rule) {
         .checkUnchanged();
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5035">[CALCITE-5035]
+   * Pull up constant project under sort</a>. */
+  @Test void testSortProjectPullUpConstants1() {
```

Comment:
You have different test cases for a reason, try to highlight what is the specificity in the name of the test instead of using `1`, `2` etc.

It will help debug test failures, but it will also make the life of other people easier to understand if what they are looking for is already handled (tests are documentation too, and their names are a big part of it).

For instance, here the focus looks like it's the absence of fetch/limit, maybe we can rename it as `testSortProjectPullUpConstantsNoFetchLimit`?

The same comment applies for the other tests.